pkg/cmd/stack/deploy.go

```diff
@@ -129,21 +129,63 @@ func deploy(ctx context.Context, s *model.Stack, c kubernetes.Interface, config
 			return
 		}
 
-		for _, name := range options.ServicesToDeploy {
-			if len(s.Services[name].Ports) > 0 {
-				svcK8s := translateService(name, s)
+		for _, serviceName := range options.ServicesToDeploy {
+			// for each service to deploy - check if there are public ports to deploy service
+			if len(s.Services[serviceName].Ports) > 0 {
+				svcK8s := translateService(serviceName, s)
 				if err := services.Deploy(ctx, svcK8s, c); err != nil {
 					exit <- err
 					return
 				}
-				ingressPorts := getSvcPublicPorts(name, s)
-				for _, ingressPort := range ingressPorts {
-					ingressName := name
-					if len(ingressPorts) > 1 {
-						ingressName = fmt.Sprintf("%s-%d", name, ingressPort.ContainerPort)
+				// get the public ports from the compose service - this will be deployed into ingresses
+				ingressPortsToDeploy := getSvcPublicPorts(serviceName, s)
+				for _, ingressPort := range ingressPortsToDeploy {
+					ingressName := serviceName
+					// If more than one port, ingressName will have <serviceName>-<PORT>, each port will have an ingress
+					if len(ingressPortsToDeploy) > 1 {
+						ingressName = fmt.Sprintf("%s-%d", serviceName, ingressPort.ContainerPort)
 					}
-					svcIngress := translateSvcIngress(ingressName, name, ingressPort.ContainerPort, s)
-					if err := deployIngress(ctx, svcIngress, iClient, spinner); err != nil {
+					translateOptions := &ingresses.TranslateOptions{
+						Name:      s.Name,
+						Namespace: s.Namespace,
+					}
+
+					// merge endpoint labels and stack specific labels
+					endpointLabels := map[string]string{
+						model.StackNameLabel:         s.Name,
+						model.StackEndpointNameLabel: ingressName,
+					}
+					for k, v := range s.Services[serviceName].Labels {
+						endpointLabels[k] = v
```

Comment:
Is it ok for this to be able to override the labels above?

Reply:
ok, i've refactored this part to have clearer logic